File: src/transformers/features/protobuf/ProtobufCreateDecodeTransformer.ts
import { ProtobufDecodeProgrammer } from "../../../programmers/protobuf/ProtobufDecodeProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufCreateDecodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "protobuf.createDecode",
      write: ProtobufDecodeProgrammer.write,
    });
}

File: packages/typia/native/transform/features/protobuf/ProtobufCreateDecodeTransformer.go
package protobuf

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativeprotobufprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/protobuf"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type protobufCreateDecodeTransformerNamespace struct{}

var ProtobufCreateDecodeTransformer = protobufCreateDecodeTransformerNamespace{}

func (protobufCreateDecodeTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Factory(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "protobuf.createDecode",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativeprotobufprogrammers.ProtobufDecodeProgrammer.Write(nativeprotobufprogrammers.ProtobufDecodeProgrammer_IProps(x))
		},
	})
}
